vite.config.js

```diff
@@ -1,12 +1,120 @@
-import { defineConfig } from 'vite';
-import solidPlugin from 'vite-plugin-solid';
+import { defineConfig } from "vite";
+import { VitePWA } from "vite-plugin-pwa";
+import solidPlugin from "vite-plugin-solid";
+import { dependencies } from './package.json';
+
+function renderChunks(deps) {
+  let chunks = {};
+  Object.keys(deps).forEach((key) => {
+    if (['@deriv/deriv-api', 'solid-js'].includes(key)) return;
+    chunks[key] = [key];
+  });
+  return chunks;
+}
 
 export default defineConfig({
-  plugins: [solidPlugin()],
+  plugins: [
+    solidPlugin(),
+    VitePWA({
+      base: "/",
+      strategies: 'generateSW',
+      registerType: 'autoUpdate',
+      workbox: {
+        globPatterns: ['index.html'],
+        runtimeCaching: [
+          {
+            urlPattern: /.*(js|css|ico|png|html|svg)$/,
+            handler: "CacheFirst",
+            options: {
+              cacheName: "project-files",
+              expiration: {
+                maxAgeSeconds: 60 * 60 * 24,
+              },
+            },
+          },
+        ],
+      },
+      manifest: {
```

Comment:
will the same work if we use `manifestFile` property and specify the path of the manifest file?